### PR TITLE
perf(gatsby): Only polyfill node modules if IE support is needed

### PR DESCRIPTION
**Pull request by j0sh77**
## Description

We currently run babel polyfilling on (most) node_modules for browser support. Realistically, this is only needed for IE. We should use the browserlist given to us by a site's `package.json` to determine whether or not we should polyfill these packages. 

In some cases, we saw the `Building production JavaScript and CSS bundles` step go from 230s to 140s, a 39% speed increase for this step. This leaves the choice up to the developer on whether or not IE support is desired, in exchange for build speed improvements. 

[sc-50607]

### CODE REVIEW

File: packages/gatsby/src/utils/browserslist.ts [hunk 1]
```diff
@@ -36,3 +36,8 @@ export const hasES6ModuleSupport = (directory: string): boolean => {
   const browserslist = getBrowsersList(directory)
   return query(browserslist + `, not supports es6-module`).length === 0
 }
+
+export const hasIESupport = (directory: string): boolean => {
+  const browserslist = getBrowsersList(directory)
+  return query(browserslist + ` and ie > 1`).length > 0
```
Comment (wardpeet) [inline packages/gatsby/src/utils/browserslist.ts:42]: Maybe es6-module is a better fit. es6-module roughly checks of all es6 features like Array.from
```suggestion
  return query(browserslist + ` and supports es6-module`).length > 0
```

File: packages/gatsby/src/utils/webpack-utils.ts [hunk 2]
```diff
@@ -512,11 +512,14 @@ export const createWebpackUtils = (
         `warning`,
         `webpack`,
       ]
-      const doNotPolyfillRegex = new RegExp(
-        `[\\\\/]node_modules[\\\\/](${VENDORS_TO_NOT_POLYFILL.join(
-          `|`
-        )})[\\\\/]`
-      )
+
+      const doNotPolyfillRegex = hasIESupport(process.cwd())
```
Comment (LekoArts) [inline packages/gatsby/src/utils/webpack-utils.ts:516]: As per making moves, we'll want to move this `hasIESupport` check up one level and don't even use the loader if not applicable

File: packages/gatsby/src/utils/browserslist.ts (hunk 1 above)
Comment (j0sh77) [reply packages/gatsby/src/utils/browserslist.ts]: Good call. That actually simplified this diff quite a bit.

**Comment by j0sh77** [comment]
> `build-javascript polyfills dependencies when ES6 module support is false` seems to fail correctly in CI

Huh, this is weird. I haven't been able to find what part of setup we're doing things differently, but the test should be pretty explicit in it's setup - stage being `build-javascript` and the mocked ES6 support being false. I'll set the mock in the test as well just to make sure.

**Comment by j0sh77** [comment]
Looks like it's an issue with the windows environment. There aren't any env vars here that should affect the test run, so I'm not sure what else could be affecting this. Going to have to dig a bit more.

**Review by LekoArts — APPROVED** [review]

**Pull request merged**